### PR TITLE
Adds the court agent role

### PR DESCRIPTION
**Pull request by FreeStylaLT**
## About The Pull Request
Adds the Court Agent role, available only in lobby / roundstart.
They are beholden to the Hand first, and the Keep hierarchy second.
2 slots.
Requires 5 PQ.

It's identical to an Adventurer in every way, except:

- Readying as it will not have your character show up in the lobby.
dreamseeker_9ej2wzrvib

- the Hand, upon arriving / spawning, will get a list of any agents in the game:
dreamseeker_RKOvmZ4Zv3

- Every Garrison and Keep role will be able to identify court agents on examine:
dreamseeker_v9GY7nFC5p

(This is what a non-Garrison and non-Keep role would see):
dreamseeker_wg63TfVpFh

## Why It's Good For The Game
This was mostly done by request. This should be safe to TM.

The general idea was to give the Hand sneaky shady agents to use.

### CODE REVIEW

**Comment by kittysmooch** [comment]
frumentarii real

**Comment by Onutsio** [comment]
I dont rlly think we need a court intelligence role when the hand can just hire rogues which is kinda neat rp. But I'm sort of eh abt it either way

**Comment by Dakken243** [comment]
> I dont rlly think we need a court intelligence role when the hand can just hire rogues which is kinda neat rp. But I'm sort of eh abt it either way

I think it's decent as in general as hand most don't seem interested in being hired for that sort of work, only like 1-2 people I've seen period inrp. Part of it I feel is people not knowing/thinking of it, and part of it that people are uninterested.

**Comment by ShudderEve** [comment]
A really nice twist to something that's already being done on a regular basis! The Hand is meant to be a strategist, and a resource of intrigue, I think allowing them to have few people out there sided with them would enable a lot of RP, and give people a reason to play Hand more!

**Comment by radarseas** [comment]
I'd give it a few more options like pilgrims, vagabonds, towners and mercenaries.

nobody expects that chill smith that's been handing you gear for scraps to also be the guy that's been keeping tabs on your names, numbers, gear and whatever you're up to, or where the scrap was stolen from.

or the random """harmless""" beggar that's been watching you break into the sewer entry to the keep for the 230th time to be a direct agent to the crown.

**Comment by FreeStylaLT** [comment]
> Love this! Really leans into the 'spymaster hand' gimmick and gives the keep a dedicated rogue-esque role - which is something I feel like the keep's been lacking for a while.

To note, at the moment the role is a straight up Adventurer, meaning that they have all the relevant subclasses to pick from. So they may not always be a rogue.

**Comment by PPooch** [comment]
Maybe give the Hand a "recruit Agent" spell/ability, like for example how the Sergeant can recruit town watch? That way there could be use of it mid-round, for hiring rogues and the like.

**Comment by dwasint** [comment]
> Maybe give the Hand a "recruit Agent" spell/ability, like for example how the Sergeant can recruit town watch? That way there could be use of it mid-round, for hiring rogues and the like.

I did this in my pr on vanderlin, except its a scroll and learning of the agents is done via reading the scrolls so their is an incentive to steal it.

File: code/modules/mob/dead/dead.dm [hunk 1]
```diff
@@ -108,11 +108,15 @@ INITIALIZE_IMMEDIATE(/mob/dead)
 			else
 				PL2 += "[PL[i]], "
 
+		var/str_job = job.title
+		if(job.title == "Court Agent")
+			str_job = "Adventurer"
```
Comment (dwasint) [inline code/modules/mob/dead/dead.dm:113]: this will cause two instances of adventurers to appear in the list

Comment (FreeStylaLT) [reply code/modules/mob/dead/dead.dm:113]: Guh, didn't notice that.

I guess at this point it'd be simpler to just not have them show up in the lobby at all.

Comment (radarseas) [reply code/modules/mob/dead/dead.dm:113]: probably a better choice, its not like anyone's going to pay attention to 20+ names, _plus early latejoins exist at the roundstart mark to blend in as plasuable deniable meta anyway._

**Pull request merged**